pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -255,22 +262,20 @@ func (d *defaultDriver) Reconcile(ctx context.Context) *reconciler.Results {
 			msg := "Could not reconcile cluster license, re-queuing"
 			log.Info(msg, "err", err, "namespace", d.ES.Namespace, "es_name", d.ES.Name)
 			d.ReconcileState.AddEvent(corev1.EventTypeWarning, events.EventReasonUnexpected, fmt.Sprintf("%s: %s", msg, err.Error()))
-			d.ReconcileState.UpdateElasticsearchState(*resourcesState, observedState())
-			results.WithResult(defaultRequeue)
+			results.WithReconciliationState(defaultRequeue.WithReason(msg))
 		}
 	}
 
 	// reconcile remote clusters
 	if esReachable {
 		requeue, err := remotecluster.UpdateSettings(ctx, d.Client, esClient, d.Recorder(), d.LicenseChecker, d.ES)
+		msg := "Could not update remote clusters in Elasticsearch settings, re-queuing"
```

Comment:
Do we not also requeue to update the remote cluster annotation after deletion of remote clusters from the Elasticsearch settings. In that case the message "could not update remote clusters" would be incorrect because we were able to update the settings but we need an extra iteration to sync the annotation.

Reply:
Yes, you're right 👍 . Fixed in https://github.com/elastic/cloud-on-k8s/commit/5e99e35d1dfe840bb365f8ce7354eb6a46c0f0bd